src/Compilers/CSharp/Test/Semantic/Semantics/ParamsSpanTests.cs

```diff
@@ -0,0 +1,1448 @@
+                Diagnostic(ErrorCode.ERR_OvlUnaryOperatorExpected, "(A a, params ReadOnlySpan<B> args)").WithLocation(8, 38),
+                // (8,44): error CS1670: params is not valid in this context
+                //     public static implicit operator B(A a, params ReadOnlySpan<B> args)
+                Diagnostic(ErrorCode.ERR_IllegalParams, "params").WithLocation(8, 44));
+        }
+
+        /// <summary>
+        /// Optional parameters are allowed before 'params'.
+        /// </summary>
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OptionalParameters()
+        {
+            var sourceA =
+@"using System;
+public class A
+{
+    public static void F1(int x, int y = 2, params Span<int> args)
+    {
+        Console.WriteLine((x, y));
+        foreach (var arg in args) Console.WriteLine(arg);
+    }
+    public static void F2<T>(T x = default, params ReadOnlySpan<T> args)
+    {
+        Console.WriteLine(x);
+        foreach (var arg in args) Console.WriteLine(arg);
+    }
+}";
+            var comp = CreateCompilation(sourceA, targetFramework: TargetFramework.NetCoreApp);
+            comp.VerifyDiagnostics();
+            var refA = comp.EmitToImageReference();
+
+            var sourceB =
+@"class B
+{
+    static void Main()
+    {
+        A.F1(1);
+        A.F1(2, 3);
+        A.F1(3, 4, 5);
+        A.F2<int>();
+        A.F2(1);
+        A.F2(2, 3, 4);
+    }
+}";
+            CompileAndVerify(sourceB, references: new[] { refA }, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, verify: Verification.Skipped, expectedOutput:
+@"(1, 2)
+(2, 3)
+(3, 4)
+5
+0
+1
+2
+3
+4
+");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void IArgumentOperation_01()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F<T>(params Span<T> args) { }
+    static void Main()
+    {
+        /*<bind>*/F(1, 2)/*</bind>*/;
+    }
+}";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp);
+            VerifyOperationTreeAndDiagnosticsForTest<SyntaxNode>(
+                comp,
+@"    IInvocationOperation (void Program.F<System.Int32>(params System.Span<System.Int32> args)) (OperationKind.Invocation, Type: System.Void) (Syntax: 'F(1, 2)')
+  Instance Receiver:
+    null
+  Arguments(1):
+      IArgumentOperation (ArgumentKind.ParamArray, Matching Parameter: args) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: 'F(1, 2)')
+        IOperation:  (OperationKind.None, Type: System.Span<System.Int32>, IsImplicit) (Syntax: 'F(1, 2)')
+          Children(4):
+              ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Span<System.Int32>, IsImplicit) (Syntax: 'F(1, 2)')
+                Left:
+                  ILocalReferenceOperation:  (OperationKind.LocalReference, Type: System.Span<System.Int32>, IsImplicit) (Syntax: 'F(1, 2)')
```

Comment:
>ILocalReferenceOperation:  (OperationKind.LocalReference, Type: System.Span<System.Int32>, IsImplicit) (Syntax: 'F(1, 2)') [](http://example.com/codeflow?start=18&length=122)

What is this local? I do not see any locals in the body. I Operation tree shouldn't contain any locals synthesized by lowering.